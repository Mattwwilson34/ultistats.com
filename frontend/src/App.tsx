import React from 'react'
import { Button } from '@mui/material'
import { auth } from '../../firebase/firebase'

function App(): React.ReactElement {
  const handleLogout = (): void => {
    void (async () => {
      try {
        await auth.signOut()
      } catch (error) {
        console.log(error)
      }
    })()
  }
  return (
    <div className="App" data-testid="App">
      <Button variant="text">{`Welcome`}</Button>
      <Button variant="contained" onClick={handleLogout}>
        Logout
      </Button>
    </div>
  )
}

export default App
