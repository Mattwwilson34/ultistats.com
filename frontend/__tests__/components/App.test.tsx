import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import App from '../../src/App'

describe('App Component', () => {
  test('should render the App component correctly', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<App />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
