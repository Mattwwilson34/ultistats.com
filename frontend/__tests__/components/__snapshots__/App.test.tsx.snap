// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the App component correctly 1`] = `
<div
  className="App"
  data-testid="App"
>
  <TopAppNav />
  <TeamList
    teams={
      [
        "Atlanta Hustle",
        "Austin Sol",
        "Chicago Union",
        "Dallas Roughnecks",
      ]
    }
  />
</div>
`;
